torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2537,6 +2567,57 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
   }
 }
 
+void ProcessGroupNCCL::allgatherUniqueNCCLID(
+    int rootRank,
+    ncclUniqueId* ncclID,
+    std::vector<ncclUniqueId>& ncclIDs) {
+  std::vector<std::string> storeKeys;
+  for (size_t r = 0; r < ncclIDs.size(); r++) {
+    storeKeys.emplace_back("UniqueNCCLID:" + std::to_string(r));
+  }
+  if (rank_ == rootRank) {
+    auto vec = std::vector<uint8_t>(
+        reinterpret_cast<uint8_t*>(ncclID),
+        reinterpret_cast<uint8_t*>(ncclID) + NCCL_UNIQUE_ID_BYTES);
+    store_->set(storeKeys[rootRank], vec);
+  }
+  try {
+    int indx = 0;
+    for (auto& value : store_->multiGet(storeKeys)) {
+      TORCH_CHECK_WITH(
+          DistBackendError,
+          value.size() == NCCL_UNIQUE_ID_BYTES,
+          "Invalid size for ncclUniqueId");
+      std::memcpy(&ncclIDs[indx++], value.data(), value.size());
+    }
+  } catch (const std::exception& e) {
```

Comment:
nit: shall we only wrap `store_->multiGet` with try-catch? 
The other part of the code does not seem relevant to the handling in catch.